src/app/checkout/accordion.controller.js

```diff
@@ -650,7 +572,62 @@ angular.module('checkoutModule')
             });
         }
 
-        function applyDiscount() {
+        $scope._sendPostForm = function(method, response) {
+            var form = "<div class='hidden' id='auth_net_form'>" + response.result + '</div>';
+            form = form.replace('$CC_NUM', method.cc.number);
+            form = form.replace('$CC_MONTH', method.cc.expire_month);
+            form = form.replace('$CC_YEAR', method.cc.expire_year);
+
+            $('body').append(form);
+            $('#auth_net_form').find('form').submit();
+            $('#auth_net_form').remove();
+        };
+
+        $scope._isValid = function() {
+            var message = '',
+                result = {
+                status: true,
+                message: '',
+            };
+
+            for (var step in $scope._isValidSteps) {
+                if ($scope._isValidSteps.hasOwnProperty(step) && !$scope._isValidSteps[step]) {
+                    message += $scope._getErrorMsg(step);
+                    result = {
+                        status: false,
+                        message: message
+                    };
+                }
+            }
+
+            return result;
+        };
+
+        $scope._getErrorMsg = function(step) {
+            /*jshint maxcomplexity:6 */
```

Comment:
We really should work to not have functions with a cyclomatic complexity over 5 and strive for a typical max complexity of 3 in most scenarios.